File: api-frontend/apps/web-antd/src/api/system/log.ts
import { requestClient } from '#/api/request';

export interface OperationLogRecord {
  id: number;
  tenant_id: number;
  user_id: number;
  username: string;
  module: string;
  action: string;
  method: string;
  path: string;
  ip: string;
  user_agent: string;
  params: string;
  result: string;
  status: number; // 1-成功 0-失败
  error: string;
  duration: number;
  created_at: string;
}

export interface LoginLogRecord {
  id: number;
  tenant_id: number;
  user_id: number;
  username: string;
  ip: string;
  user_agent: string;
  login_type: string; // password, totp
  status: number; // 1-成功 0-失败
  error: string;
  created_at: string;
}

export interface PageResult<T> {
  list: T[];
  total: number;
  page: number;
  page_size: number;
}

// 操作日志
export function getOperationLogList(params: {
  page: number;
  page_size: number;
  keyword?: string;
  module?: string;
  status?: number;
  start_time?: string;
  end_time?: string;
}) {
  return requestClient.get<PageResult<OperationLogRecord>>('/logs/operation', { params });
}

// 登录日志
export function getLoginLogList(params: {
  page: number;
  page_size: number;
  keyword?: string;
  status?: number;
  start_time?: string;
  end_time?: string;
}) {
  return requestClient.get<PageResult<LoginLogRecord>>('/logs/login', { params });
}

export interface EmailLogRecord {
  id: number;
  tenant_id: number;
  to: string;
  subject: string;
  status: number;
  error: string;
  created_at: string;
}

export interface SmsLogRecord {
  id: number;
  tenant_id: number;
  phone: string;
  template_id: string;
  params: string;
  status: number;
  error: string;
  created_at: string;
}

// 邮件日志
export function getEmailLogList(params: {
  page: number;
  page_size: number;
  keyword?: string;
}) {
  return requestClient.get<PageResult<EmailLogRecord>>('/logs/email', { params });
}

// 短信日志
export function getSmsLogList(params: {
  page: number;
  page_size: number;
  keyword?: string;
}) {
  return requestClient.get<PageResult<SmsLogRecord>>('/logs/sms', { params });
}

export interface LogConfig {
  log_operation_enabled: string;
  log_login_enabled: string;
  log_email_enabled: string;
  log_sms_enabled: string;
}

// 日志配置
export function getLogConfig() {
  return requestClient.get<LogConfig>('/configs/log');
}

export function updateLogConfig(data: LogConfig) {
  return requestClient.put('/configs/log', data);
}

File: api-backend/internal/model/log.go
package model

import "time"

type OperationLog struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	TenantID  uint      `gorm:"index" json:"tenant_id"`
	UserID    uint      `gorm:"index" json:"user_id"`
	Module    string    `gorm:"size:50" json:"module"`
	Action    string    `gorm:"size:50" json:"action"`
	Method    string    `gorm:"size:10" json:"method"`
	Path      string    `gorm:"size:255" json:"path"`
	Params    string    `gorm:"type:text" json:"params"`
	Response  string    `gorm:"type:text" json:"response"`
	IP        string    `gorm:"size:45" json:"ip"`
	UserAgent string    `gorm:"size:500" json:"user_agent"`
	Duration  int64     `json:"duration"`
	CreatedAt time.Time `json:"created_at"`
}

func (OperationLog) TableName() string {
	return "operation_logs"
}

type LoginLog struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	TenantID  uint      `gorm:"index" json:"tenant_id"`
	UserID    uint      `gorm:"index" json:"user_id"`
	Username  string    `gorm:"size:50" json:"username"`
	IP        string    `gorm:"size:45" json:"ip"`
	UserAgent string    `gorm:"size:500" json:"user_agent"`
	Status    int8      `json:"status"`
	Message   string    `gorm:"size:255" json:"message"`
	CreatedAt time.Time `json:"created_at"`
}

func (LoginLog) TableName() string {
	return "login_logs"
}

type EmailLog struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	TenantID  uint      `gorm:"index" json:"tenant_id"`
	To        string    `gorm:"size:255" json:"to"`
	Subject   string    `gorm:"size:500" json:"subject"`
	Status    int8      `json:"status"` // 1=成功 0=失败
	Error     string    `gorm:"size:500" json:"error"`
	CreatedAt time.Time `json:"created_at"`
}

func (EmailLog) TableName() string {
	return "email_logs"
}

type SmsLog struct {
	ID         uint      `gorm:"primarykey" json:"id"`
	TenantID   uint      `gorm:"index" json:"tenant_id"`
	Phone      string    `gorm:"size:20" json:"phone"`
	TemplateID string    `gorm:"size:50" json:"template_id"`
	Params     string    `gorm:"size:500" json:"params"`
	Status     int8      `json:"status"` // 1=成功 0=失败
	Error      string    `gorm:"size:500" json:"error"`
	CreatedAt  time.Time `json:"created_at"`
}

func (SmsLog) TableName() string {
	return "sms_logs"
}
